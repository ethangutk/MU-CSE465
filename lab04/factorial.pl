% Compute the factorial function 
%   n! = n*(n-1)!
%   0! = 1

%% TODO -- Put your implementation here
fact(0, 1).
fact(N, F) :- M is N - 1, fact(M, G), F is N*G.


%% Testing

:- fact(5, X), format('5! is ~d~n', X), !.
:- fact(0, X), format('0! is ~d~n', X), !.
:- fact(1, X), format('1! is ~d~n', X), !.
:- fact(3, X), format('3! is ~d~n', X), !.
:- fact(7, X), format('7! is ~d~n', X), !.
