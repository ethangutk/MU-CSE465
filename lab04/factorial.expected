5! is 120
0! is 1
1! is 1
3! is 6
7! is 5040
